hub/core/dataset.py

```diff
@@ -204,18 +214,53 @@ def create_tensor(
             htype=htype,
             dtype=dtype,
             sample_compression=sample_compression,
+            hash_samples=hash_samples,
             **meta_kwargs,
         )
-        self.meta.tensors.append(name)
+        
         self.storage.maybe_flush()
         tensor = Tensor(name, self.storage)  # type: ignore
-
-        self.tensors[name] = tensor
+        
+        if hash_samples:
+            hashes_tensor = self.create_tensor(HASHES_TENSOR_FOLDER, htype="hashes")
+            self._link_tensor(tensor, hashes_tensor)
+           
+        # Hashes tensor is considered as a "hidden tensor"
+        if tensor.meta.htype == "hashes":
+            self.meta.hidden_tensors.append(name)
+            self.hidden_tensors[name] = tensor
```

Comment:
can we not always hide hash htype tensors? if im a user trying to create my own, it will hide it and i will be very confused.

i recommend adding a separate function called `_create_hidden_tensor`

Reply:
@mccrearyd or would it be better to include it as a parameter in create_tensor i.e `create_tensor('hashes', htype='hash', hidden_tensor=True')`? The tensor meta file will also have an additional `hidden_tensor` field